cmd/x-check-mailserver/main.go

```diff
@@ -0,0 +1,241 @@
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"os"
+	stdsignal "os/signal"
+	"strconv"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/status-im/go-ethereum/common"
+	"github.com/status-im/status-go/logutils"
+	"github.com/status-im/status-go/params"
+	"github.com/status-im/status-go/signal"
+	"github.com/status-im/statusd-bots/protocol"
+	"github.com/status-im/whisper/shhclient"
+	whisper "github.com/status-im/whisper/whisperv6"
+)
+
+func init() {
+	if err := logutils.OverrideRootLog(true, *verbosity, "", false); err != nil {
+		log.Fatalf("failed to override root log: %v\n", err)
+	}
+}
+
+func main() {
+	// handle OS signals
+	signals := make(chan os.Signal, 1)
+	stdsignal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+
+	go func() {
+		<-signals
+		os.Exit(1)
+	}()
+
+	// create config
+	config, err := newNodeConfig(*fleet, params.MainNetworkID)
+	if err != nil {
+		log.Fatalf("failed to create a config: %v", err)
+	}
+
+	// collect mail servers
+	mailserversToCheck := *mailservers
+	if len(mailserversToCheck) == 0 {
+		// -2 to get at least two mail servers
+		min := rand.Intn(len(config.ClusterConfig.TrustedMailServers) - 1)
+		max := min + rand.Intn(len(config.ClusterConfig.TrustedMailServers)-min)
+		mailserversToCheck = config.ClusterConfig.TrustedMailServers[min:max]
+	}
+
+	config.MaxPeers = len(mailserversToCheck)
+
+	// collect mail server request signals
+	mailSignalsForwarder := newSignalForwarder()
+	defer close(mailSignalsForwarder.in)
+	go mailSignalsForwarder.Start()
+
+	// setup signals handler
+	signal.SetDefaultNodeNotificationHandler(
+		filterMailTypesHandler(printHandler, mailSignalsForwarder.in),
+	)
+
+	// setup work
+	workConfig := WorkUnitConfig{
+		From:    uint32(time.Now().Add(-*duration).Unix()),
+		To:      uint32(time.Now().Unix()),
```

Comment:
i think it needs to exclude last minutes of envelopes. otherwise the chance of being inconsistent will be very high, the same as in https://github.com/status-im/status-react/issues/6814

Reply:
Good point!